### PR TITLE
Hooks

### LINKED ISSUE #845: Migrate to Hooks
The code presently uses class components & lifecycle methods.
However, React 16.8.0 release introduces a more intuitive & functional approach towards development which is more likely to be used in the future.

## Steps to resolve the issue:
Replace Class and Class-based methods (state, lifecycle, PureComponent etc.) to functional approach (useState, useEffect, memo, etc.)

## Keep in mind:
https://github.com/OperationCode/front-end/issues/867

## Note

Comment the names of the 5 components you plan on migrating before tackling them so you don't step on anybodys' toes.

### PR DESCRIPTION
**Pull request by LeviButcher**
# Description of changes

Update class components to hooks

* LoginForm
* RegistrationForm
* PasswordReset
* code_schools
* _error

# Issue Resolved

Fixes #845

### CODE REVIEW

**Comment by kylemh** [comment]
🤖 **This is a bot** 🤖<br /><br />🎉 Deployed Storybook preview! 🎉<br /><br />Click the link at the bottom of this comment to see it live.

Built with commit 81efe18db19fb10b40216ff415901d965c3aa24f

https://deploy-preview-885--operation-code-storybook.netlify.com

**Comment by kylemh** [comment]
#### 📝Changed routes:


| [**/_error**](https://operation-code-git-fork-levibutcher-hooks.operation-code.now.sh/_error) | [**/code_schools**](https://operation-code-git-fork-levibutcher-hooks.operation-code.now.sh/code_schools) |
|:-:|:-:|
|<a href="https://operation-code-git-fork-levibutcher-hooks.operation-code.now.sh/_error">Screenshot of /_error</a><br /><sup><a href="https://microlink-cdn.s3.amazonaws.com/s/v6HJDG8nPepGI3/1zusO_gc2Z9Tw_">(view full size)</a> |<a href="https://operation-code-git-fork-levibutcher-hooks.operation-code.now.sh/code_schools">Screenshot of /code_schools</a><br /><sup><a href="https://microlink-cdn.s3.amazonaws.com/s/BqegjnXFskyCYh/HTp6nX4W62v8E_">(view full size)</a> |







Commit 81efe18db19fb10b40216ff415901d965c3aa24f (https://operation-code-git-fork-levibutcher-hooks.operation-code.now.sh).

File: pages/code_schools.js [hunk 1]
```diff
@@ -38,278 +39,268 @@ const moocSchools = [
...
     const vetTecApprovedSchools = allSchools.filter(school => school.isVetTecApproved);
-    this.setState({ filteredSchools: vetTecApprovedSchools, selectedStates: [] });
+    setFilterSchools(vetTecApprovedSchools);
+    setSelectedStates([]);
   };
 
-  showAllSchools = () => {
-    const { allSchools } = this.props;
-    this.setState({ filteredSchools: allSchools, selectedStates: [] });
+  const showAllSchools = () => {
```
Comment (kylemh) [inline pages/code_schools.js]: Can you define this near the top of the method so calls are below the definition?

Comment (LeviButcher) [reply pages/code_schools.js]: Yup, no problem. :+1:

**Pull request merged**